extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -413,20 +413,30 @@ export class AmpAnalytics extends AMP.BaseElement {
    * @param {string} template The template to expand.
    * @param {!JSONObject} The object to use for variable value lookups.
    * @param {!Object} event Object with details about the event.
+   * @param {number} opt_iterations Number of recursive expansions to perform.
+   *    Defaults to 2 substitutions.
    * @return {string} The expanded string.
    * @private
    */
-  expandTemplate_(template, trigger, event) {
+  expandTemplate_(template, trigger, event, opt_iterations) {
+    opt_iterations = opt_iterations === undefined ? 2 : opt_iterations;
+    if (opt_iterations < 0) {
+      return '';
```

Comment:
Yeah, I saw the tests, but not sure I fully understand this. E.g. why not just return `12${3}` instead of `12`?

This is probably totally fine, but I just want to get a better picture...